lib/MetaCPAN/Server/Action/Deserialize.pm

```diff
@@ -10,9 +10,10 @@ around serialize_bad_request => sub {
 
     $c->res->status(400);
 
+    $error =~ s/^(.*) at [^\s]+ line .*$/$1/mi;
+
     my $full_error
         = "Content-Type $content_type had a problem with your request.\n$error";
-    $full_error =~ s{ at .*? line \d+\.\n\z}{};
```

Comment:
This change seems incorrect. There shouldn't be any reason to use a case insensitive match, nor does it make sense to remove the location from the first line of the error rather than the end of the string.

The original is also slightly wrong and should probably be `s{ at \S+ line \d+\.\n\z}{};`.

Reply:
Thank you for taking your time to review my work.
I agree with you about the _ignorecare_ flag in the RegEx instruction.
The text it is working on is generated by the _Perl_ Interpreter so it is not expected to change.
In this sense I reviewed the instruction and simplified it.
The change of location of this instruction was to add **Predictability** and **Maintainability** to the code since the instruction actually tries to cleanse the **incoming Error Message** in `$error` rather than correcting the resulting Error Message in `$full_error`. It adds predictability because it avoids influences of later introduces **New Line Characters** `\n` in `$full_error`. In that way it will help later following Colleagues to easier understand and debug the code.

Also you can be sure that this Expression was crafted after carefully study of the Data to process.

Nevertheless I checked your proposed instruction with:
```
--- a/lib/MetaCPAN/Server/Action/Deserialize.pm
+++ b/lib/MetaCPAN/Server/Action/Deserialize.pm
@@ -10,10 +10,13 @@ around serialize_bad_request => sub {
 
     $c->res->status(400);
 
-    $error =~ s/^(.*) at [^\s]+ line .*$/$1/mi;
+    print "err 0 dmp\n'$error'\n";
+
+    #$error =~ s/^(.*) at [^\s]+ line .*$/$1/mi;
 
     my $full_error
         = "Content-Type $content_type had a problem with your request.\n$error";
+    $full_error =~ s{ at \S+ line \d+\.\n\z}{};
 
     $c->stash(
         {
diff --git a/t/server/controller/bad_request.t b/t/server/controller/bad_request.t
index 3611077b..faa84176 100644
--- a/t/server/controller/bad_request.t
+++ b/t/server/controller/bad_request.t
@@ -56,6 +56,7 @@ for (@tests) {
                     'Content-type'
                 );
                 my $content = eval { decode_json( $res->content ) };
+                print "cntnt dmp:\n'" . $res->content ; print "'\n";
                 ok( is_hashref($content), 'response is a JSON object' );
                 ok $content->{$field}, "response includes field '$field'";
                 ok $content->{$field} =~ /$check/i , 'response error message as expected';
```
Unfortunately I found that the proposed Expression **does not match** the Error Message:
```
# Subtest: broken body content
    # Subtest: check with 'release' controller
err 0 dmp
'unexpected end of string while parsing JSON string, at character offset 18 (before "(end of string)") at /usr/local/lib/perl5/site_perl/5.30.1/Catalyst/Action/Deserialize/JSON.pm line 35, <$fh> line 1.
'
        ok 1 - GET /release/_search
        ok 2 - code [400] as expected
        ok 3 - Content-type
cntnt dmp:
'{
   "error" : "Content-Type application/json had a problem with your request.\nunexpected end of string while parsing JSON string, at character offset 18 (before \"(end of string)\") at /usr/local/lib/perl5/site_perl/5.30.1/Catalyst/Action/Deserialize/JSON.pm line 35, <$fh> line 1.\n"
}
'
        ok 4 - response is a JSON object
        ok 5 - response includes field 'error'
        ok 6 - response error message as expected
        1..6
    ok 1 - check with 'release' controller
    # Subtest: check with 'file' controller
err 0 dmp
'unexpected end of string while parsing JSON string, at character offset 18 (before "(end of string)") at /usr/local/lib/perl5/site_perl/5.30.1/Catalyst/Action/Deserialize/JSON.pm line 35, <$fh> line 1.
'
        ok 1 - GET /file/_search
        ok 2 - code [400] as expected
        ok 3 - Content-type
cntnt dmp:
'{
   "error" : "Content-Type application/json had a problem with your request.\nunexpected end of string while parsing JSON string, at character offset 18 (before \"(end of string)\") at /usr/local/lib/perl5/site_perl/5.30.1/Catalyst/Action/Deserialize/JSON.pm line 35, <$fh> line 1.\n"
}
'
        ok 4 - response is a JSON object
        ok 5 - response includes field 'error'
        ok 6 - response error message as expected
        1..6
    ok 2 - check with 'file' controller
    1..2
ok 1 - broken body content
# Subtest: plain text query
    # Subtest: check with 'release' controller
err 0 dmp
'malformed JSON string, neither tag, array, object, number, string or atom, at character offset 0 (before "some content as inva...") at /usr/local/lib/perl5/site_perl/5.30.1/Catalyst/Action/Deserialize/JSON.pm line 35, <$fh> line 1.
'
        ok 1 - GET /release/_search
        ok 2 - code [400] as expected
        ok 3 - Content-type
cntnt dmp:
'{
   "error" : "Content-Type application/json had a problem with your request.\nmalformed JSON string, neither tag, array, object, number, string or atom, at character offset 0 (before \"some content as inva...\") at /usr/local/lib/perl5/site_perl/5.30.1/Catalyst/Action/Deserialize/JSON.pm line 35, <$fh> line 1.\n"
}
'
        ok 4 - response is a JSON object
        ok 5 - response includes field 'error'
        ok 6 - response error message as expected
        1..6
    ok 1 - check with 'release' controller
    # Subtest: check with 'file' controller
err 0 dmp
'malformed JSON string, neither tag, array, object, number, string or atom, at character offset 0 (before "some content as inva...") at /usr/local/lib/perl5/site_perl/5.30.1/Catalyst/Action/Deserialize/JSON.pm line 35, <$fh> line 1.
'
        ok 1 - GET /file/_search
        ok 2 - code [400] as expected
        ok 3 - Content-type
cntnt dmp:
'{
   "error" : "Content-Type application/json had a problem with your request.\nmalformed JSON string, neither tag, array, object, number, string or atom, at character offset 0 (before \"some content as inva...\") at /usr/local/lib/perl5/site_perl/5.30.1/Catalyst/Action/Deserialize/JSON.pm line 35, <$fh> line 1.\n"
}
'
        ok 4 - response is a JSON object
        ok 5 - response includes field 'error'
        ok 6 - response error message as expected
        1..6
    ok 2 - check with 'file' controller
    1..2
ok 2 - plain text query
```

Still if you prefer to **continue to work** on the Expression and create another Pull Request I can **exclude** this file from this branch.

Reply:
Maybe we could exclude the regex from this PR and squash it down to one commit?

Reply:
I excluded the `MetaCPAN::Server::Action::Deserialize` module and run the `perltidy` script on the test